node_modules/oae-principals/lib/api.user.js

```diff
@@ -774,3 +809,225 @@ var _setAdmin = function(ctx, adminType, isAdmin, principalId, callback) {
         return PrincipalsDAO.setAdmin(adminType, isAdmin, principalId, callback);
     });
 };
+
+/**
+ * Send an email token to a user that can be used to verify the user owns the email address
+ *
+ * @param  {Context}    ctx             Standard context object containing the current user and the current tenant
+ * @param  {User}       user            The user to send the email token to
+ * @param  {String}     email           The email address where to send the token to
+ * @param  {String}     [token]         The token to send. If left null, a new one will be generated
+ * @param  {Function}   callback        Standard callback function
+ * @param  {Object}     callback.err    An error that occurred, if any
+ * @api private
+ */
+var _sendEmailToken = function(ctx, user, email, token, callback) {
+    callback = callback || function(err) {
+        if (err) {
+            log().error({'err': err, 'userId': user.id}, 'Unable to send a user a verification email');
+        }
+    };
+
+    // Generate a token if none was specified
+    if (!token) {
+        token = ShortId.generate();
+    }
+
+    // Store the token
+    PrincipalsDAO.storeEmailToken(user.id, email, token, function(err) {
+        if (err) {
+            return callback(err);
+        }
+
+        // The EmailAPI expects a user to have a verified email address. As this is not the case
+        // when sending an email token, we send in a patched user object
+        var userToEmail = _.extend({}, user, {'email': email});
+
+        var fullTenant = TenantsAPI.getTenant(user.tenant.alias);
+        var url = TenantsUtil.getBaseUrl(fullTenant) + '/email-verification?token=' + token;
+
+        // Send an email to the specified e-mail address
+        var data = {
+            'actor': ctx.user(),
+            'tenant': ctx.tenant(),
+            'user': userToEmail,
+            'baseUrl': TenantsUtil.getBaseUrl(ctx.tenant()),
+            'skinVariables': require('oae-ui').getTenantSkinVariables(ctx.tenant().alias),
+            'token': token,
+            'verificationUrl': url
+        };
+
+        // We pass the current date in as the "hashCode" for this email. We need to be able to send
+        // a copy of the same email for the "resend email token" functionality. As we don't expect
+        // that this logic will get stuck in a loop this is probably OK
+        EmailAPI.sendEmail('oae-principals', 'verify', userToEmail, data, {'hash': Date.now()}, callback);
```

Comment:
Just want to make sure it wasn't a missed piece of review feedback, and that you don't want to make the API take in an email as a parameter, as opposed to extend it onto the user object.